### PR TITLE
fix: check buildPath consistently

### PR DESCRIPTION
**Pull request by jcfranco**
*Issue #, if available:* #1144 

*Description of changes:*

Extracted `buildPath`-checking logic to its own util and ensure it uses the platform-specific separator.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jorenbroekema** [comment]
Not sure if I can get this into v3, @dbanksdesign this fix and [this one](https://github.com/amzn/style-dictionary/pull/1112) are waiting for your approval, might be nice to do another v3 release since v4 is still in prerelease/alpha stage.

I'll make sure to port this fix to v4, it's slightly more nuanced of a fix there since it has to be browser-compatible as well, and in browsers a filesystem shim is used that uses posix separators regardless of whether the user agent is Windows

**Review by jorenbroekema — APPROVED** [review]

**Comment by jcfranco** [comment]
@jorenbroekema Thanks for the review! 

@dbanksdesign LMK if there's anything else needed from my side to help get this landed for a v3 release. Thanks!

**Comment by jorenbroekema** [comment]
v4 is stable and merged to main by now, so closing this as fixed in v4

**Pull request closed without merging**
